clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs

```diff
@@ -0,0 +1,4405 @@
+import qualified Data.Text           as Text
+import           GHC.Exts (IsList(..))
+import           GHC.Float
+import           GHC.Int
+import           GHC.Integer
+  (decodeDoubleInteger,encodeDoubleInteger,compareInteger,orInteger,andInteger,
+   xorInteger,complementInteger,absInteger,signumInteger)
+import           GHC.Integer.GMP.Internals
+  (Integer (..), BigNat (..))
+import           GHC.Natural
+import           GHC.Prim
+import           GHC.Real            (Ratio (..))
+import           GHC.TypeLits        (KnownNat)
+import           GHC.Types           (IO (..))
+import           GHC.Word
+import           System.IO.Unsafe    (unsafeDupablePerformIO)
+
+import           BasicTypes          (Boxity (..))
+import           Name                (getSrcSpan, nameOccName, occNameString)
+import           PrelNames
+  (typeNatAddTyFamNameKey, typeNatMulTyFamNameKey, typeNatSubTyFamNameKey,
+   trueDataConKey, falseDataConKey)
+import           SrcLoc              (wiredInSrcSpan)
+import qualified TyCon
+import           TysWiredIn          (tupleTyCon)
+import           Unique              (getKey)
+
+import           Clash.Class.BitPack (pack,unpack)
+import           Clash.Core.DataCon  (DataCon (..))
+import           Clash.Core.Evaluator.Types
+import           Clash.Core.Literal  (Literal (..))
+import           Clash.Core.Name
+  (Name (..), NameSort (..), mkUnsafeSystemName)
+import           Clash.Core.Pretty   (showPpr)
+import           Clash.Core.Term
+  (Pat (..), PrimInfo (..), Term (..), WorkInfo (..), mkApps)
+import           Clash.Core.TermInfo (piResultTys)
+import           Clash.Core.Type
+  (Type (..), ConstTy (..), LitTy (..), TypeView (..), mkFunTy, mkTyConApp,
+   splitFunForallTy, tyView)
+import           Clash.Core.TyCon
+  (TyConMap, TyConName, tyConDataCons)
+import           Clash.Core.TysPrim
+import           Clash.Core.Util
+  (mkRTree,mkVec,tyNatSize,dataConInstArgTys,primCo,
+   undefinedTm)
+import           Clash.Core.Var      (mkLocalId, mkTyVar)
+import           Clash.Debug         (trace)
+import           Clash.GHC.GHC2Core  (modNameM)
+import           Clash.Rewrite.Util  (mkSelectorCase)
+import           Clash.Unique        (lookupUniqMap)
+import           Clash.Util
+  (MonadUnique (..), clogBase, flogBase, curLoc)
+
+import Clash.Promoted.Nat.Unsafe (unsafeSNat)
+import qualified Clash.Sized.Internal.BitVector as BitVector
+import qualified Clash.Sized.Internal.Signed    as Signed
+import qualified Clash.Sized.Internal.Unsigned  as Unsigned
+import Clash.Sized.Internal.BitVector(BitVector(..), Bit(..))
+import Clash.Sized.Internal.Signed   (Signed   (..))
+import Clash.Sized.Internal.Unsigned (Unsigned (..))
+import Clash.XException (isX)
+
+import {-# SOURCE #-} Clash.GHC.Evaluator
+
+-- | Evaluation of primitive operations.
+-- TODO This should really be in Clash.GHC.Evaluator -- the evaluator in
+-- clash-lib should NEVER refer to GHC primitives.
```

Comment:
:-)

Reply:
Moving a file not magically changing the comments I left in it? Inconceivable!